src/main/java/com/thevoxelbox/voxelsniper/sniper/toolkit/BlockTracer.java

```diff
@@ -1,43 +1,23 @@
 package com.thevoxelbox.voxelsniper.sniper.toolkit;
 
-import org.bukkit.Location;
-import org.bukkit.block.Block;
+import com.sk89q.worldedit.bukkit.BukkitAdapter;
+import com.sk89q.worldedit.math.BlockVector3;
 import org.bukkit.entity.Player;
-import org.bukkit.util.BlockIterator;
-
-import java.util.Iterator;
+import org.jetbrains.annotations.Nullable;
 
 public class BlockTracer {
 
-    private Block targetBlock;
-    private Block lastBlock;
+    private final BlockVector3 targetBlock;
 
-    public BlockTracer(Player player, int distance) {//FAWE modify (make public)
-        Location eyeLocation = player.getEyeLocation();
-        Block block = eyeLocation.getBlock();
-        this.targetBlock = block;
-        this.lastBlock = block;
-        Iterator<Block> iterator = new BlockIterator(player, distance);
-        iterate(iterator);
-    }
+    public BlockTracer(Player player, int distance, boolean useLastBlock) {//FAWE modify (make public)
```

Comment:
We don't really need these annotations here anymore, or never really used them. Upstream VS is dead, can remove them on sight imo.

Reply:
👍